### ISSUE TITLE
play 2.8.8 : JDK 16 class play.utils.Resources$ cannot access class sun.net.www.protocol.file.FileURLConnection

### ISSUE THREAD

**Issue by larousso** [playframework/playframework#10838]
Running play with the jdk 16 crash when request hit an enpoint using assets with the following error : 

```
java.lang.IllegalAccessError: class play.utils.Resources$ (in unnamed module @0x36174d32) cannot access class sun.net.www.protocol.file.FileURLConnection (in module java.base) because module java.base does not export sun.net.www.protocol.file to unnamed module @0x36174d32
	at play.utils.Resources$.isUrlConnectionADirectory(Resources.scala:36)
	at controllers.AssetsBuilder.$anonfun$assetAt$3(Assets.scala:828)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at play.api.libs.streams.Execution$trampoline$.executeScheduled(Execution.scala:109)
	at play.api.libs.streams.Execution$trampoline$.execute(Execution.scala:71)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.dispatchOrAddCallback(Promise.scala:316)
	at scala.concurrent.impl.Promise$DefaultPromise.onComplete(Promise.scala:307)
	at scala.concurrent.impl.Promise.transformWith(Promise.scala:40)
	at scala.concurrent.impl.Promise.transformWith$(Promise.scala:38)
	at scala.concurrent.impl.Promise$DefaultPromise.transformWith(Promise.scala:187)
	at scala.concurrent.Future.flatMap(Future.scala:306)
	at scala.concurrent.Future.flatMap$(Future.scala:306)
	at scala.concurrent.impl.Promise$DefaultPromise.flatMap(Promise.scala:187)
	at play.api.libs.streams.Accumulator$.$anonfun$futureToSink$1(Accumulator.scala:231)
	at akka.stream.impl.Transform.apply(TraversalBuilder.scala:159)
	at akka.stream.impl.PhasedFusingActorMaterializer.materialize(PhasedFusingActorMaterializer.scala:522)
	at akka.stream.impl.PhasedFusingActorMaterializer.materialize(PhasedFusingActorMaterializer.scala:448)
	at akka.stream.impl.PhasedFusingActorMaterializer.materialize(PhasedFusingActorMaterializer.scala:440)
	at akka.stream.scaladsl.RunnableGraph.run(Flow.scala:744)
	at play.api.libs.streams.SinkAccumulator.run(Accumulator.scala:139)
	at play.api.libs.streams.SinkAccumulator.run(Accumulator.scala:140)
	at play.api.libs.streams.FlattenedAccumulator.$anonfun$run$2(Accumulator.scala:216)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:63)
	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:100)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:100)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:49)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:48)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:295)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by mkurz** [comment]
Play 2.8 supports "only" Java 8 and Java 11.
However,  I am working to make (not released) Play 2.9 support Java 17.
BTW: The problem that occured to you already got fixed in #10731

**Comment by 5tefan** [comment]
For macOS users w/ brew:

```
brew install java11
brew link --overwrite java11
```

seems to make brew installed sbt and scala use java11.
